client/src/pages/forms/SignupForm.js

```diff
@@ -0,0 +1,162 @@
+import React from "react";
+import { useState, useEffect } from "react";
+
+import "./Form.css";
+
+
+function SignupForm() {
+	// eslint-disable-next-line react-hooks/exhaustive-deps
+	const initialDetails = {
+		Name: "",
+		Class: "",
+		Username: "",
+		Password: "",
+		ConfirmPassword: "",
+	};
+	const [details, setDetails] = useState(initialDetails);
+	const [errors, setErrors] = useState({});
+	const [submit, setSubmit] = useState(false);
+
+	function submitHandler(e) {
+		e.preventDefault();
+		setErrors(validate(details));
+		setSubmit(true);
+	}
+	useEffect(() => {
+		console.log("errors", errors);
+		if (Object.keys(errors).length === 0 && submit) {
+			setDetails(initialDetails);
+			console.log("details entered:", details);
+		}
+	}, [details, errors, initialDetails, submit]);
+
+	const validate = (details) => {
+		const errors = {};
+		const userNameRegex = /^[a-z0-9]+(?:[ _.-][a-z0-9]+)*$/;
+
+
+		if (!details.Name) {
+			errors.Name = "Name is required";
+		} else if (details.Name.length < 3) {
+			errors.Name = "Please enter your full name";
+		}
+		if (!details.Class) {
+			errors.Class = "Class is required";
+		} else if (details.Class.length < 3 || details.Class.length > 7) {
+			errors.Class = "Please enter the correct Class. i.e: WM3, NW4";
+		}
+		if (!details.Username) {
+			errors.Username = "Username is required";
+		} else if (
+			details.Username.length < 3 ||
+			!userNameRegex.test(details.Username)
+		) {
+			errors.Username = "Required format: word, word-sep-word";
+		}
+		if (!details.Password) {
+			errors.Password = "Password is required";
+		} else if (details.Password.length < 3) {
+			errors.Password = "Password is too short";
+		}
+		if (details.ConfirmPassword !== details.Password) {
+			errors.ConfirmPassword = "Passwords do not match";
+		}
+
+		return errors;
+	};
+	return (
+		<section className="signup__forms">
+			<div className="trainee__form">
+
+				{Object.keys(errors).length === 0 && submit ? (
+					<div className="ui msg success">Signed Up Successfully</div>
+				) : (
+					""
+				)}
+				<form onSubmit={submitHandler}>
+					<div className="form-inner">
+						<h2>Sign Up Page</h2>
+						<h3>Please enter your details as a trainee</h3>
+
+						<div className="form-group">
+							<label htmlFor="trainee_name">Name:</label>
+							<input
+								type="text"
+								name="trainee_name"
+								id="trainee_name"
+								onChange={(e) =>
+									setDetails({ ...details, Name: e.target.value })
+								}
+								value={details.Name}
+							/>
+						</div>
+						<p className="form__error">{errors.Name}</p>
+						<div className="form-group">
+							<label htmlFor="trainee_name">Class:</label>
+							<input
+								type="text"
```

Comment:
so this here would be a dropdown that's populated via fetching from the /cohorts endpoint in the backend

Reply:
made dropdown menu but haven't done the fetching yet